### DISCUSSION TITLE
Commands for AMD and generic possible optimizations

### DISCUSSION THREAD

**Discussion by LeGitHubs** [vladmandic/sdnext#902] [Q&A]
I've put **set COMMANDLINE_ARGS= --medvram --use-directml** in webui.bat

I found some other online: --opt-sdp-attention --opt-split-attention  --opt-sub-quad-attention --disable-nan-check --medvram --always-batch-cond-uncond --no-half-vae --precision-full

But these don't work when added. : error: unrecognized arguments... Press any key to continue . . . 

Using gradio ui theme, windows 10,  Have:Amd Radeon RX 5700 Xt 50 Anniversary & amd ryzen 9 3950x 16-Core Processor

Any other working commands I should try? Or setting names in the settings after launching.

-----------------------------------------------------------------------------
Possible Optimization stuff (maybe something is useful to implement if not already done)
-----------------------------------------------------------------------------

https://www.reddit.com/r/StableDiffusion/comments/102uatr/mcmc_sampling_speeds_up_diffusion_models/

Using ram disc: 30% faster? Maybe could be added to Vlad in an already setup way somehow? https://www.reddit.com/r/StableDiffusion/comments/ycbqju/odd_way_to_speed_up_stable_diffusion_gui_its_10/
" In Linux, you can use Kexec and PMEM to make a block device from RAM. You can load your whole OS into it along with SD to load stupid fast.
https://gist.github.com/noblepayne"

https://www.reddit.com/r/StableDiffusion/comments/xmr3ic/speed_up_stable_diffusion_by_50_using_flash/

Another but they using already fast card and output different:https://www.reddit.com/r/StableDiffusion/comments/z1kw35/upto_25x_speed_up_of_stablediffusiondreambooth/

Any AMD optimizations that could be learned from Shark since I heard it runs fast for amd: https://github.com/nod-ai/SHARK
old stat [from somebody else nothing to do with my specs]
Shark:
4.0s/it (prompt "test" with 50 steps takes around 8.5s)

WebUI-DirectML:
1.3s/it (prompt "test" with 50 steps takes around 65s)

**Comment by OldPixelReaper** [comment]
You can set many of the arguments in the WebUI in the settings.

--medvram --use-directml should be enough.  BTW, why does it run at 1.3 it/s for you with the same graphics card I have? It was the same for me, but with the new build it always switches to the CPU for me and I'm at 24 it/s

**Reply by LeGitHubs** [reply]
Sorry for writing confusing for that. That's not my seconds. Just a old comparison I found for the shark thing from somebody else.

Atm making a 512x512 image for me takes some minutes. 

goes from 8 eta to 9 eta to 6 eta to 7 eta then 5 eta going up and down 4 eta etc

Steps: 20
Time taken: 4m 21.57s

Some people are rolling back to a previous commit maybe this could fix it for you. 

https://github.com/vladmandic/automatic/discussions/732

To go to a previous commit (as far as a I know, learned today)

Copy folder adress
C:\Users\YOUR FOLDER NAME HERE\Vlad1111\automatic
Change it to this format in a notepad: cd /C/Users/YOUR FOLDER NAME HERE/Vlad1111/automatic

open gitbash (search on computer if on windows)
paste 
cd /C/Users/lolus/Vlad1111/automatic


git checkout 5148c5b

**Comment by vladmandic** [comment]
VoltaML is based on TensorRT and that requires a completely different backend and models to be recompiled for it (and compiling a model is not really on-the-fly thing).
Shark is similar.

And using RAM disk is old idea, but it doesn't help when I have 100+ models on my disk. Plus loading a model from a fast SSD is matter of seconds (3-5 sec on mine).

For Linux users Torch + AMD ROCm is a decent solution. I wish that AMD put more effort into their ROCm library (like nVidia is putting into CUDA) as its definitely sub-par. And it doesn't (yet) exists on Windows. So for Windows at the moment, fallback to Torch-DirectML is only viable solution that would be compatible with the rest of the ecosystem, but its faaar from ideal (it basically needs to do two round trips to do same amount of work). If you want a single model model, no options, no extras, no extensions, etc - then yes, you can explore TensorRT, Shark, VoltaML, etc.

**Reply by Aptronymist** [reply]
> VoltaML is based on TensorRT and that requires a completely different backend and models to be recompiled for it (and compiling a model is not really on-the-fly thing). Shark is similar.
> 
> And using RAM disk is old idea, but it doesn't help when I have 100+ models on my disk. Plus loading a model from a fast SSD is matter of seconds (3-5 sec on mine).
> 
> For Linux users Torch + AMD ROCm is a decent solution. I wish that AMD put more effort into their ROCm library (like nVidia is putting into CUDA) as its definitely sub-par. And it doesn't (yet) exists on Windows. So for Windows at the moment, fallback to Torch-DirectML is only viable solution that would be compatible with the rest of the ecosystem, but its faaar from ideal (it basically needs to do two round trips to do same amount of work). If you want a single model model, no options, no extras, no extensions, etc - then yes, you can explore TensorRT, Shark, VoltaML, etc.

I'm actually wondering about this since you've implemented directml now it seems, I now have both an amd gpu and an nvidia gpu, can I force it to use the amd card with --use-directml?
The way it reads is that it will only do that if it can't find a cuda card, which it will find.

**Reply by vladmandic** [reply]
yes, i've added that yesterday.
